patcher/src/components/Sound/index.tsx

```diff
@@ -39,40 +40,52 @@ const sounds = {
 export class Sound extends React.Component<SoundProps, SoundState> {
 
   private bgRef: HTMLAudioElement = null;
+  private evhs: any[] = [];       // event handlers
   private audioRefs: { [id: string]: HTMLAudioElement } = {};
 
   constructor(props: SoundProps) {
     super(props);
     this.state = {
       sounds: {},
+      paused: false,
     };
   }
 
   public render() {
     return (
       <div>
-        <audio src='sounds/patcher-theme.ogg' ref={r => this.bgRef = r }/>
-        {this.renderAudioElements()}     
+        <audio loop={true}
+               src='sounds/patcher-theme.ogg'
+               ref={r => this.bgRef = r }/>
+        {this.renderAudioElements()}
       </div>
     );
   }
 
+  private setStateAsync = (sparseState: any) => {
+    setTimeout(() => { this.setState(sparseState); },0);
```

Comment:
To prevent errors about state being modified during render.  

``PatchButton:render()`` ultimately calls ``events.fire()`` for ``pause-music`` which in turn fires off the handler in the Sound component, which wants to modify state (``SoundState``) to update the paused state, and for some reason best know to ``React`` it complains that ``setState`` is being called during ``render``.